profiler/src/ProfilerEngine/Datadog.Profiler.Native/ContentionProvider.cpp

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+#include "ContentionProvider.h"
+
+#include "COMHelpers.h"
+#include "IAppDomainStore.h"
+#include "IFrameStore.h"
+#include "IManagedThreadList.h"
+#include "IRuntimeIdStore.h"
+#include "IThreadsCpuManager.h"
+#include "OsSpecificApi.h"
+#include "Sample.h"
+
+ContentionProvider::ContentionProvider(
+    ICorProfilerInfo4* pCorProfilerInfo,
+    IManagedThreadList* pManagedThreadList,
+    IFrameStore* pFrameStore,
+    IThreadsCpuManager* pThreadsCpuManager,
+    IAppDomainStore* pAppDomainStore,
+    IRuntimeIdStore* pRuntimeIdStore)
+    :
+    CollectorBase<RawContentionSample>("ContentionProvider", pThreadsCpuManager, pFrameStore, pAppDomainStore, pRuntimeIdStore),
+    _pCorProfilerInfo{pCorProfilerInfo},
+    _pManagedThreadList{pManagedThreadList}
+{
+}
+
+void ContentionProvider::OnContention(double contentionDuration)
+{
+    ManagedThreadInfo* threadInfo;
+    CALL(_pManagedThreadList->TryGetCurrentThreadInfo(&threadInfo))
```

Comment:
Since there is no sampling, we might send too many logs in this case

Reply:
This is equivalent to the AllocationProvider ;)

On another thing, this would be very rare no?

Reply:
You're right: keep this one